technical-reference/reference-technical-articles/technical-doc-diamond.md

```diff
@@ -0,0 +1,80 @@
+## Overview:
+
+The problem at hand arose when additional features were introduced to the Comptroller contract within the core pool, causing it to surpass the maximum allowable size of 24KB. In response to this challenge and to preemptively address similar issues in the future, a multifaceted solution was implemented.
+
+The solution involves the incorporation of the EIP-2535 pattern into the Comptroller contract, which has been refactored into distinct facets. This restructuring aligns the storage layout with that of the Comptroller proxy, streamlining the contract's organization and enhancing efficiency. Consequently, user interactions now trigger two delegate calls, optimizing the contract's performance and addressing the size constraint effectively.
```

Comment:
> optimizing the contract's performance

How does the use of two delegatecall optimize the performance?

Reply:
Removed ` optimizing the contract's performance`.